### PR TITLE
Feat(cupertino): [date_picker] Date order parameter

### LINKED ISSUE #84550: [proposal] provide a way to change the date picker order in the `CupertinoDatePicker` constructor
## Use case

I need to change the order of a specific `CupertinoDatePicker`. It currently defaults to:

Screen Shot 2021-06-14 at 09 09 01

## Proposal

Provide a way to change the order of a `CupertinoDatePicker` during its creation (in the constructor). So that I can have something similar to:

Screen Shot 2021-06-14 at 09 10 07

### PR DESCRIPTION
**Pull request by ayshiff**
Closes #84550 

This PR adds a new `dateOrder` parameter to  `CupertinoDatePicker` to define the order of the columns in date mode which overrides the default order value defined by `localizations.datePickerDateOrder`.

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/test/cupertino/date_picker_test.dart
Comment (pedromassango) [inline packages/flutter/test/cupertino/date_picker_test.dart:1248]: Is this last line required? I mean the assignment is not being used.

Maybe you meant to remove it or add another `expect` statement.

**Review by pedromassango — CHANGES_REQUESTED** [review]
LGTM once the comments are addressed

File: packages/flutter/lib/src/cupertino/date_picker.dart
Comment (pedromassango) [inline packages/flutter/lib/src/cupertino/date_picker.dart:1064]: Nit: space between

**Review by pedromassango — CHANGES_REQUESTED** [review]
One more nit and should be go to go

**Review by pedromassango — APPROVED** [review]
LGTM

I would like to get a final approval from @LongCatIsLooong to make sure we're not missing anything.

File: packages/flutter/test/cupertino/date_picker_test.dart
Comment (LongCatIsLooong) [inline packages/flutter/test/cupertino/date_picker_test.dart:1247]:
```suggestion
      expect(
        tester.getTopLeft(find.text('14')).dx,
        lessThan(tester.getTopLeft(find.text('January')).dx,
      );
```

File: packages/flutter/lib/src/cupertino/date_picker.dart
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/cupertino/date_picker.dart:250]: `[localizations]` does not exist I think? Maybe just say defaults to the locale's default date format/order?

**Comment by pedromassango** [comment]
@ayshiff could you please rebase for the CI to re-run?

**Pull request merged**
